### PR TITLE
Add http-types dependency

### PR DESCRIPTION
**Pull request by matycarolina**
Add http-types dependency to library

### CODE REVIEW

File: src/Shortener.hs [hunk 1]
```diff
@@ -1,43 +1,51 @@
...
 import Network.HTTP.Types (status404)
 import Text.Blaze.Html.Renderer.Text (renderHtml)
 import qualified Text.Blaze.Html5 as H
 import qualified Text.Blaze.Html5.Attributes as A
+import Text.RE.Replace
+import Text.RE.TDFA.String
 import Web.Scotty
 
+onlyLetters :: String -> String
+onlyLetters s = replaceAll "" $ s *=~ [re|$([^a-zA-Z])|]
```
Comment (jpvillaisaza) [inline src/Shortener.hs]: Can you add an example of what the function does? What's the output for `https://github.com/`?

Comment (matycarolina) [reply src/Shortener.hs]: Using this function, the output would be  `httpsgithubcom` but that's not what we intended to do. The goal was to validate the URL in order to make sure the user inputs a valid URL.

File: src/Shortener.hs [hunk 2]
```diff
@@ -2,42 +2,54 @@
...
       url <- param "url"
-      liftIO $ modifyIORef urlsR $
-        \(i, urls) ->
-          (i + 1, M.insert i url urls)
+      ( if isValid (unpack url)
+          then liftIO $
+            modifyIORef urlsR $
+              \(i, urls) ->
+                (i + 1, M.insert i url urls)
+          else raiseStatus status404 "not valid"
```
Comment (jpvillaisaza) [inline src/Shortener.hs]: I think this shouldn't return a 404 (not found).

Comment (matycarolina) [reply src/Shortener.hs]: yes, agreed. Maybe a 400 (bad request) would be a better fit, since it would indicate that the client did something wrong.

**Review by jpvillaisaza — APPROVED** [review]

**Pull request merged**